pkg/controller/node/nodecontroller.go

```diff
@@ -368,7 +369,18 @@ func (nc *NodeController) maybeDeleteTerminatingPod(obj interface{}) {
 
 func forcefullyDeletePod(c clientset.Interface, pod *api.Pod) {
 	var zero int64
-	err := c.Core().Pods(pod.Namespace).Delete(pod.Name, &api.DeleteOptions{GracePeriodSeconds: &zero})
+	serverPod, err := c.Core().Pods(pod.Namespace).Get(pod.Name)
+	if kubeerrors.IsNotFound(err) {
+		glog.V(3).Infof("Pod %q (%s) does not exist on the server", serverPod.Name, serverPod.UID)
+		return
+	}
+	// This doesn't entirely avoid the race condition, pod can be deleted and recreated after this check
+	// but before the deletion request sent to the server.
```

Comment:
Can we do that for all pod operations? In kubelet, we also have to check if the pod we're getting from the apiserver before a status update is actually the pod we mean to update. The same applies to deletion. Since we use the pod UID as the key, it seems to make sense to be able to send a request to the apiserver with UID.